tests/accessor/accessor_api_image_common.h

```diff
@@ -1104,16 +1104,26 @@ class image_accessor_api_sampled_r {
     const bool useNormalized =
         m_sampler.coordinate_normalization_mode ==
             sycl::coordinate_normalization_mode::normalized;
+    const bool useNearest =
+        m_sampler.filtering_mode == sycl::filtering_mode::nearest;
+
     if (useNormalized) {
       const bool worksForLower =
         check_read<acc_coord_tag::use_normalized_lower>(idx);
       if (worksForLower)
         check_read<acc_coord_tag::use_normalized_upper>(idx);
     } else {
-      const bool worksForInteger =
-        check_read<acc_coord_tag::use_int>(idx);
-      if (worksForInteger)
+      const bool worksForFloat =
         check_read<acc_coord_tag::use_float>(idx);
+      if (worksForFloat && useNearest) {
+        /*
+          OpenCL spec rules that when int is used as the coordinates, the
```

Comment:
We should reference SYCL spec, not OpenCL.

Reply:
Good point.